lib/charms/identity_platform_login_ui_operator/v0/login_ui_endpoints.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Interface library for sharing Identity Platform Login UI application's endpoints with other charms.
+This library provides a Python API for both requesting and providing a public endpoints.
+## Getting Started
+To get started using the library, you need to fetch the library using `charmcraft`.
+```shell
+cd some-charm
+charmcraft fetch-lib charms.identity_platform_login_ui.v0.login_ui_endpoints
+```
+To use the library from the requirer side:
+In the `metadata.yaml` of the charm, add the following:
+```yaml
+requires:
+  ui-endpoints-info:
+    interface: login_ui_endpoints
+    limit: 1
+```
+Then, to initialise the library:
+```python
+from charms.identity_platform_login_ui.v0.login_ui_endpoints import (
+    LoginUIEndpointsRelationError,
+    LoginUIEndpointsRequirer,
+)
+Class SomeCharm(CharmBase):
+    def __init__(self, *args):
+        self.login_ui_endpoints_relation = LoginUIEndpointsRequirer(self)
+        self.framework.observe(self.on.some_event_emitted, self.some_event_function)
+    def some_event_function():
+        # fetch the relation info
+        try:
+            login_ui_endpoints = self.login_ui_endpoints_relation.get_login_ui_endpoints()
+        except LoginUIEndpointsRelationError as error:
+            ...
+```
+"""
+
+import logging
+from typing import Dict, Optional
+
+from ops.charm import CharmBase, RelationCreatedEvent
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+
+# The unique Charmhub library identifier, never change it
+LIBID = "460ab09e6b874d1c891b67f83586c9a7"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 3
```

Comment:
LIBPATCH should be set to `1` since this is the first version

Reply:
done